src/annotator/anchoring/pdf.js

```diff
@@ -448,55 +480,51 @@ export function anchor(root, selectors) {
  * @param {HTMLElement} root - The root element
  * @param {Range} range
  */
-export function describe(root, range) {
-  const normalizedRange = new BrowserRange(range).normalize();
+export async function describe(root, range) {
+  // "Shrink" the range so that the start and endpoints are at offsets within
+  // text nodes rather than any containing nodes.
+  try {
+    range = TextRange.fromRange(range).toRange();
+  } catch {
+    throw new Error('Selection does not contain text');
+  }
 
-  const startTextLayer = getNodeTextLayer(normalizedRange.start);
-  const endTextLayer = getNodeTextLayer(normalizedRange.end);
+  const startTextLayer = getNodeTextLayer(range.startContainer);
+  const endTextLayer = getNodeTextLayer(range.endContainer);
 
-  if (startTextLayer !== endTextLayer) {
-    return Promise.reject(
-      new Error('selecting across page breaks is not supported')
-    );
+  if (!startTextLayer || !endTextLayer) {
+    throw new Error('Selection is outside page text');
   }
 
-  const startRange = normalizedRange.limit(startTextLayer);
-  const endRange = normalizedRange.limit(endTextLayer);
-
-  if (!startRange || !endRange) {
-    return Promise.reject(new Error('range is outside text layer'));
+  if (startTextLayer !== endTextLayer) {
+    throw new Error('Selecting across page breaks is not supported');
   }
 
-  const startPageIndex = getSiblingIndex(startTextLayer.parentNode);
+  const startPos = TextPosition.fromPoint(
+    range.startContainer,
+    range.startOffset
+  ).relativeTo(startTextLayer);
+
+  const endPos = TextPosition.fromPoint(
+    range.endContainer,
+    range.endOffset
+  ).relativeTo(endTextLayer);
 
-  const iter = root.ownerDocument.createNodeIterator(
-    startTextLayer,
-    NodeFilter.SHOW_TEXT
+  const startPageIndex = getSiblingIndex(
+    /** @type {Node} */ (startTextLayer.parentNode)
   );
-  let startPos = seek(iter, normalizedRange.start);
-  let endPos =
-    seek(iter, normalizedRange.end) +
-    startPos +
-    normalizedRange.end.textContent.length;
-
-  return getPageOffset(startPageIndex).then(pageOffset => {
-    startPos += pageOffset;
-    endPos += pageOffset;
-
-    const position = new TextPositionAnchor(
-      root,
-      startPos,
-      endPos
-    ).toSelector();
-
-    const quoteRange = document.createRange();
-    quoteRange.setStartBefore(startRange.start);
-    quoteRange.setEndAfter(endRange.end);
-
-    const quote = TextQuoteAnchor.fromRange(root, quoteRange).toSelector();
-
-    return Promise.all([position, quote]);
-  });
+  const pageOffset = await getPageOffset(startPageIndex);
+
+  /** @type {TextPositionSelector} */
+  const position = {
+    type: 'TextPositionSelector',
+    start: pageOffset + startPos.offset,
+    end: pageOffset + endPos.offset,
+  };
+
+  const quote = TextQuoteAnchor.fromRange(root, range).toSelector();
+
+  return Promise.all([position, quote]);
```

Comment:
Is this wrapping in `Promise.all` because the calling code expects to get a `thenable` or something? Or is `quote` somehow async in a way I'm not spotting (totally possible given the code paths required to get to its value and it being late-ish in my day)?

Reply:
Good question! - Actually this is a mistake. The `Promise.all` is a hangover from the previous code which supported generating the position and quote selectors concurrently, even though there wasn't any performance benefit. It isn't needed any more. I'll remove it.